app_dart/bin/validate_task_ownership.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:yaml/yaml.dart';
+
+/// Remote check based on flutter `repo` and the commit `ref`.
+///
+/// This currently supports `flutter/flutter` only.
+Future<List<String>> remoteCheck(String repo, String ref) async {
+  final HttpClient client = HttpClient();
+  final Uri ciYamlUrl = Uri.https('raw.githubusercontent.com', 'flutter/$repo/$ref/.ci.yaml');
+  final HttpClientRequest ciYamlRequest = await client.getUrl(ciYamlUrl);
+  final HttpClientResponse ciYamlResponse = await ciYamlRequest.close();
+  final Uri testOwnersUrl = Uri.https('raw.githubusercontent.com', 'flutter/$repo/$ref/TESTOWNERS');
+  final HttpClientRequest testOwnersRequest = await client.getUrl(testOwnersUrl);
+  final HttpClientResponse testOwnersResponse = await testOwnersRequest.close();
+  if (ciYamlResponse.statusCode != HttpStatus.ok) {
+    throw HttpException('HTTP ${ciYamlResponse.statusCode}: $ciYamlUrl');
+  }
+  if (testOwnersResponse.statusCode != HttpStatus.ok) {
+    throw HttpException('HTTP ${testOwnersResponse.statusCode}: $testOwnersUrl');
+  }
+  final String ciYamlContent = await utf8.decoder.bind(ciYamlResponse).join();
+  final String testOwnersContent = await utf8.decoder.bind(testOwnersResponse).join();
+  client.close(force: true);
+
+  final List<String> noOwnerBuilders = validateOwnership(ciYamlContent, testOwnersContent);
+  return noOwnerBuilders;
+}
+
+/// Local check is based on paths to the local `.ci.yaml` and `TESTOWNERS` files.
+List<String> localCheck(String ciYamlPath, String testOwnersPath) {
+  final File ciYamlFile = File(ciYamlPath);
+  final File testOwnersFile = File(testOwnersPath);
+  if (!ciYamlFile.existsSync() || !testOwnersFile.existsSync()) {
+    print('Make sure ciYamlPath and testOwnersPath exist.');
+    exit(1);
+  }
+  final List<String> noOwnerBuilders =
+      validateOwnership(ciYamlFile.readAsStringSync(), testOwnersFile.readAsStringSync());
+  return noOwnerBuilders;
+}
+
+/// Validate test ownership defined in `testOwnersContenct` for tests configured in `ciYamlContent`.
+List<String> validateOwnership(String ciYamlContent, String testOwnersContenct) {
+  final List<String> noOwnerBuilders = <String>[];
+  final YamlMap ciYaml = loadYaml(ciYamlContent) as YamlMap;
+  final YamlList yamlList = ciYaml['targets'] as YamlList;
+  final List<YamlMap> targets = yamlList.map<YamlMap>((dynamic target) => target as YamlMap).toList();
+  for (final YamlMap target in targets) {
+    final String builder = target['builder'] as String;
+    final String owner = getTestOwner(builder, getTypeForBuilder(builder, ciYaml), testOwnersContenct);
+    print('$builder: $owner');
+    if (owner == null) {
+      noOwnerBuilders.add(builder);
+    }
+  }
+  return noOwnerBuilders;
+}
+
+/// Validates task ownership.
+///
+/// It expects two parameters for remote validation: the flutter `repo` and the `commit`.
+///
+/// It expects three parameters for remote validation: `local` arg, the full path to the config
```

Comment:
```suggestion
/// It expects three parameters for local validation: `local` arg, the full path to the config
```

Reply:
What is the use case for local check?

Reply:
It would be useful/handy to validate changes locally when in development.

Reply:
the original suggestion is not addressed yet

Reply:
Sorry missed here. Updated.